### DISCUSSION TITLE
Who is snliu?

### DISCUSSION THREAD

**Discussion by emirgra** [neovim/neovim#29084] [General]
Who is snliu?

Hello,

I recently wanted to download neovim to use it.

wget https://github.com/neovim/neovim/releases/latest/download/nvim-linux64.tar.gz

![nvim_latest_redirection_snliu](https://github.com/neovim/neovim/assets/34007081/8de0cd32-9028-42e2-b840-7aae8466b184)

When I looked at the output of this process, I noticed that there were two redirects.

Since I said "latest" in the first redirect link, it directed me to the latest stable version. TRUE.
The second redirect led to the file uploaded to github by another user. Wrong.

![nvim_latest_redirection_snliu_16408992](https://github.com/neovim/neovim/assets/34007081/3e995f47-0aa6-4748-a460-4c58465e650f)

https://x.com/herrcore/status/1772988192678969567

I remembered this video and immediately looked at the referred user.

ID number enclosed in a red square.
repo_id = 16408992.

https://api.github.com/user/16408992

When I looked from here, I saw that the person was snliu.

![snliu_account](https://github.com/neovim/neovim/assets/34007081/1a73f1ef-7c4a-43f8-b99b-9c4e6b128a24)

So who is this snliu? Why are we downloading the neovim file uploaded by this user?

Regards.

**Comment by zeertzjq** [comment]
Why are you looking up repo ID under `user/`? The video didn't say anything about this.

**Comment by emirgra** [comment]
When we try the process in the video, a link appears. When this link is examined, the repo_id parameter is seen. This parameter is the id of the github user who uploaded the file. You can try.

If you go to this url `https://api.github.com/user/{ID}`, the account of the user with this ID will be seen.
If you go to `https://api.github.com/users/{USERNAME}` or this url, public information about the user will be seen.

It seems that this user uploaded the release files. This is the part I'm curious about anyway.

**Comment by zeertzjq** [comment]
When I try the process in this repo the `repo_id` is 16408992. When I try the process in my own repo the `repo_id` doesn't have an existing user ID, so I don't think it is a user ID.

**Comment by emirgra** [comment]
If you download the neovim file with wget as I showed above, the repo_id will appear. But for example, I uploaded pictures above. When you click on those images, you will see a url that continues like `https://private-user-images.githubusercontent.com/34007081/334867478-3e99...` here `34007081` is my account id. So if you do the process in the video `https://private-user-images.githubusercontent.com/` the parameter that comes after here is the ID of the person who uploaded the file. for example `https://private-user-images.githubusercontent.com/34007081/334867478-3e99...`. That is, my ID since I uploaded the file. `https://private-user-images.githubusercontent.com/{ID}/334867478-3e99...`

**Comment by zeertzjq** [comment]
So… How does that explain why you are looking up the `repo_id` under `user/`?

**Comment by emirgra** [comment]
Because we can see who uploaded the program to github. When we check the ID of the person who installed this program, we can see the username. If neovim had uploaded this repo there would be no problem. because it is a reliable source. But it is interesting that someone who has 2 repos and has not been active for a long time uploads it.

**Reply by clason** [reply]
No, you are not getting the point. User IDs and repo IDs are two completely different numbers. **You are checking the wrong thing so you are getting a nonsense answer.** 

Specifically, this

> When this link is examined, the repo_id parameter is seen. This parameter is the id of the github user who uploaded the file

is a **wrong** assumption.

**Comment by emirgra** [comment]
My God. This is my fault. Yes, it's totally my fault. The id of 16408992 is not a user but the id of the repo (neovim/neovim). I'm sorry for taking up your time.

**Reply by KillTheMule** [reply]
Don't be sorry please. Recent events have shown us that at least more diligence is needed, and I for one am thankful that you took a close look at the supply chain here. Even better everything's fine after all ;)
